cogeo_mosaic/backend/base.py

```diff
@@ -0,0 +1,30 @@
+"""cogeo_mosaic.backend.base: base Backend class."""
+
+import abc
+import contextlib
+
+
+class BaseBackend(metaclass=contextlib.AbstractContextManager):
+    @abc.abstractmethod
+    def __enter__(self, *args, **kwargs):
+        """Load resource"""
+
```

Comment:
definitely not an expert but I think we still need to set an `__init__`class where we can set some metadata 
we could say that each backend has an `path` or `url` for example (or all the mosaic metadata)

https://github.com/rafelafrance/traiter_vertnet/blob/fc95fbdff9ebc7ab569d24f7b3e5ee644d2c6ad6/vertnet/writers/base_writer.py#L7-L21

Also I think by default __enter__ should return self (like in ☝️ ) and we should also have a __exit__ (it won't do anything exept maybe for closing some db connection)

Reply:
Yes, both `__init__` and `__enter__` are required.

I think that by inheriting from `contextlib.AbstractContextManager` it creates a default `def __enter__: return self`: https://docs.python.org/3/library/contextlib.html#contextlib.AbstractContextManager

Reply:
It looks like `__enter__` and `__exit__` aren't created by default, so I added them to the base class.